osversion/platform_compat_windows.go

```diff
@@ -0,0 +1,70 @@
+package osversion
+
+import (
+	"strconv"
+	"strings"
+)
+
+// List of stable ABI compliant
+// ltsc releases
+// Note: List must be sorted
+var compatLtscReleases = []uint{
+	V21H2Server,
+}
+
+// Checks for stable ABI compliant OS versions.
+// Every release after WS 2022 will support the previous
+// ltsc container image. Stable ABI is in preview mode
+// for windows 11 client.
+// Refer: https://learn.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility?tabs=windows-server-2022%2Cwindows-10#windows-server-host-os-compatibility
+func CanRun(hostOSVersion, guestOSVersion string) bool {
+	// HPC containers do not have the hostOSVersion set
+	if hostOSVersion == "" {
+		return true
+	}
+
+	hostOSBuildNum := GetOSBuildNumber(hostOSVersion)
+	ctrImgBuildNum := GetOSBuildNumber(guestOSVersion)
+
+	// If host is < WS 2022, exact version match is required
+	if hostOSBuildNum < V21H2Server {
+		return strings.HasPrefix(guestOSVersion, hostOSVersion)
```

Comment:
I don't think `HasPrefix` is a good way to check this. As a degenerate case, consider `HasPrefix("500", "50")`, which returns true, but the host version 50 would _not_ be able to run guest version 500.

Reply:
I had this thought as well but saw that current windows platform matcher is just checking the prefix considering the versions coming in are not ill-formed : https://github.com/containerd/containerd/blob/main/platforms/defaults_windows.go#L52 . I agree that we need to check for major and minor version matches as well